Feature Request: ParseWeekdayError should impl `std::fmt::Display` and `std::error::Error`

I'm trying to pass in a `chrono::Weekday` using Clap v3.

```rust
struct Args {
   /// Name of the person to greet
   #[clap(short, long, parse(try_from_str = core::str::FromStr::from_str) )]
   day: chrono::Weekday,
   ...
```

But I can't, because it doesn't implement `std::fmt::Display` and `std::error::Error`

```text
error[E0277]: `ParseWeekdayError` doesn't implement `std::fmt::Display`
   --> src/main.rs:35:45
    |
35  |    #[clap(short, long, parse(try_from_str = core::str::FromStr::from_str) )]
    |                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `ParseWeekdayError` cannot be formatted with the default formatter
    |
   ::: /home/ecarroll/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/macros.rs:116:39
    |
116 |         let res = $crate::fmt::format($crate::__export::format_args!($($arg)*));
    |                                       ---------------------------------------- in this macro invocation
    |
    = help: the trait `std::fmt::Display` is not implemented for `ParseWeekdayError`
    = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
    = note: this error originates in the macro `$crate::__export::format_args` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `ParseWeekdayError: std::error::Error` is not satisfied
    --> src/main.rs:35:45
     |
35   |    #[clap(short, long, parse(try_from_str = core::str::FromStr::from_str) )]
     |                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `std::error::Error` is not implemented for `ParseWeekdayError`
     |
     = note: required because of the requirements on the impl of `From<ParseWeekdayError>` for `Box<(dyn std::error::Error + Send + Sync + 'static)>`
     = note: required because of the requirements on the impl of `Into<Box<(dyn std::error::Error + Send + Sync + 'static)>>` for `ParseWeekdayError`
note: required by a bound in `Arg::<'help>::validator`
    --> /home/ecarroll/.cargo/registry/src/github.com-1ecc6299db9ec823/clap-3.2.15/src/builder/arg.rs:1528:12
     |
1528 |         E: Into<Box<dyn Error + Send + Sync + 'static>>,
     |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `Arg::<'help>::validator`
```

--- a/src/weekday.rs
+++ b/src/weekday.rs
@@ -191,6 +191,15 @@ pub struct ParseWeekdayError {
     pub(crate) _dummy: (),
 }
 
+#[cfg(feature = "std")]
+impl std::error::Error for ParseWeekdayError {}
+
+impl fmt::Display for ParseWeekdayError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.write_fmt(format_args!("{:?}", self))
+    }
+}
+
 impl fmt::Debug for ParseWeekdayError {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         write!(f, "ParseWeekdayError {{ .. }}")

